src/device-registry/utils/insert-device-measurements.js

```diff
@@ -1,54 +1,50 @@
-const { transformMeasurements } = require("../utils/transform-measurements");
-const insertMeasurements = require("../utils/insert-measurements");
-const { doesDeviceExist } = require("../utils/does-component-exist");
-const { logObject } = require("../utils/log");
+const { transformMeasurements } = require("./transform-measurements");
+const insertMeasurements = require("./insert-measurements");
+const { doesDeviceExist } = require("./does-component-exist");
+const { logObject } = require("./log");
 
 const insertDeviceMeasurements = {
 
     // Contains modules that provide various ways to insert device measurements.
-    
-    addValuesArray: async (array_data) => {
+
+    addValuesArray: async (measurements_array) => {
 
         try {
 
-            for (let index in array_data) {
+            for (const measurement of measurements_array) {
 
-                const data = array_data[index];
+                logObject("Kafka Measurement", measurement);
 
-                const device = data.device;
-                const tenant = data.tenant;
-                const measurements = [data];
+                const device = measurement.device;
+                const tenant = measurement.tenant;
+                const measurements = [measurement];
 
                 if (!tenant || !device) {
```

Comment:
Could you consider using && here?
Why? With the | | operator, it will short circuit the comparison and accept cases where !tenant is true even the !device is false at the same time. Please review the use case and advise the intention.

Reply:
thanks @Baalmart the right operator is **&&**, I didn't think about that earlier

Reply:
@Baalmart the logic is valid basing on the intention...if device or tenant is missing, reject the measurements

Reply:
Cool, thanks.

Reply:
Hi @NoahNsimbe , the tenant is always required.....in ALL situations. There is not scenario (usecase) that is possible without a tenant. Please crosscheck further.

Reply:
@Baalmart what is your suggestion?

Reply:
Hi @NoahNsimbe , there was an earlier suggestion (&&) in this thread. But I guess in this scenario, the logic is fine because the response body is not specific. Thanks! :)